flashinfer/mla.py

```
@@ -599,13 +600,15 @@ def trtllm_batch_decode_with_kv_cache_mla(
     if isinstance(bmm2_scale, torch.Tensor):
         assert bmm2_scale.dtype == torch.float32
     if backend == "xqa":
-        if (
-            get_compute_capability(query.device)[0] != 12
-            or query.dtype != torch.float8_e4m3fn
-            or kv_cache.dtype != torch.float8_e4m3fn
-        ):
+        if not is_sm12x_supported(query.device):
+            raise ValueError("XQA MLA is only supported on SM12x GPUs")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)